extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcAuthenticationMechanism.java

```diff
@@ -0,0 +1,53 @@
+package io.quarkus.oidc.runtime;
+
+import java.util.concurrent.CompletionStage;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import io.quarkus.security.identity.IdentityProviderManager;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.vertx.http.runtime.security.ChallengeData;
+import io.quarkus.vertx.http.runtime.security.HttpAuthenticationMechanism;
+import io.vertx.ext.web.RoutingContext;
+
+@ApplicationScoped
+public class OidcAuthenticationMechanism implements HttpAuthenticationMechanism {
+
+    @Inject
+    DefaultTenantConfigResolver resolver;
+    private volatile BearerAuthenticationMechanism bearerAuth;
```

Comment:
Why are you doing this lazy creation? These objects don't have any state, there is no measurable benefit to creating them lazily, but in return you add a volatile reads to every request which is more expensive.

Reply:
Pedro was concerned about the eager creation in the constructor, so I decided to give a try, though was not entirely sure. I'll just revert. Please also see my response to your next comment.

Reply:
If you are really concerned about the object creation all that logic could just be moved to static methods as there is no state in the objects, and they no longer have to implement an interface.

Reply:
@stuartwdouglas thanks, not sure I understand :-) I guess both objects can be just created as before. There is only one extra single object in the mainstream single `application-type` case which as you said has no state and it is not possible to avoid this extra object's class be added to the native image, so it is a very tiny cost indeed.
But can you explain please what you meant about moving the logic to the static methods, what difference would it make compared to the current lazy initialization done in the instance ? `volatile` class fields would still be read...But I guess it is not what you had in mind :-)